src/intrinsics/ecma262/JSON.js

```diff
@@ -325,11 +325,7 @@ function InternalJSONClone(realm: Realm, val: Value): Value {
     // TODO #1010: NaN and Infinity must be mapped to null.
     return val;
   }
-  if (
-    (val instanceof NumberValue && !isFinite(val.value)) ||
-    val instanceof UndefinedValue ||
-    val instanceof NullValue
-  ) {
+  if (val instanceof NumberValue && !isFinite(val.value)) {
```

Comment:
There is no discussion about what issue if being resolved by this and this seems a strange PR in which to make such a change.

Reply:
The additional invariants brought out the bug with the existing serializer test suite that our cloned JSON values where not right. `undefined` doesn't get mapped to `null`. This fixes it to make the test pass. The same code change was already in #1709.